client/signup/config/flows.js

```diff
@@ -164,27 +164,12 @@ function removeUserStepFromFlow( flow ) {
 		return;
 	}
 
-	// TODO: A more general middle destination fallback mechanism is needed
-	// The `midDestination` mechanism is tied to a specific step in the flow configuration.
-	// If it happens to be the user step, then we the middle destination is also removed.
-	// For addressing Automattic/martech#1260, here we introduce a limited fallback mechanism that only works
-	// for the user step. Whenever a step that provides an auth token is removed, we simply tie the middle destination
-	// to its previous step. If it's the first step, then it will be removed all together atm.
 	const steps = [];
-	let prevStep = flow.steps[ 0 ];
-
 	for ( const curStep of flow.steps ) {
 		if ( stepConfig[ curStep ].providesToken ) {
```

Comment:
Non-blocking suggestion:
IMO, using `filter` might show the intention a bit more clearly, and might be a bit easier to read.
```js
steps = flow.steps.filter( ( stepName ) => ! stepConfig[ stepName ].providesToken );
```

Reply:
Yeah, actually that's how it used to be. I have one more follow-up clean-up PR to do, so I'll sneak it in there :)